pkg/asset/installconfig/gcp/client.go

```diff
@@ -60,28 +61,31 @@ func (c *Client) GetNetwork(ctx context.Context, network, project string) (*comp
 }
 
 // GetPublicDomains returns all of the domains from among the project's public DNS zones.
-func (c *Client) GetPublicDomains(ctx context.Context, project string) ([]string, error) {
+func (c *Client) GetPublicDomains(ctx context.Context, project string) (map[string][]string, error) {
 	ctx, cancel := context.WithTimeout(context.TODO(), 1*time.Minute)
 	defer cancel()
 
 	svc, err := c.getDNSService(ctx)
 	if err != nil {
-		return []string{}, err
+		return map[string][]string{}, err
 	}
 
-	var publicZones []string
+	publicZonesMap := map[string][]string{}
 	req := svc.ManagedZones.List(project).Context(ctx)
 	if err := req.Pages(ctx, func(page *dns.ManagedZonesListResponse) error {
 		for _, v := range page.ManagedZones {
 			if v.Visibility != "private" {
-				publicZones = append(publicZones, strings.TrimSuffix(v.DnsName, "."))
+				dnsName := strings.TrimSuffix(v.DnsName, ".")
+				publicZoneID := strconv.FormatUint(v.Id, 10)
```

Comment:
I think a more ergonomic identifier for zone is the project, zone name tuple which can be combined as `/projects/openshift-dev-installer/managedZones/installer-public-zone` or such similar to the definition https://github.com/openshift/api/blob/master/config/v1/types_dns.go#L59

Reply:
You can specify either the id or the managed zone name. My personal preference is for an ID. If you insist, I will happily change it ;-)

Reply:
> You can specify either the id or the managed zone name

if they are completely interchangeable, i think we should go with name @jhixson74 

interger identifier is something that is not seen when looking at the console or gcloud unless somebody is looking for verbose object.

Reply:
I have updated the code to use the managed zone name.